cross-chain/polygon/deploy_sidechain/02_deploy_polygon_wormhole_gateway.ts

```diff
@@ -0,0 +1,71 @@
+import type { HardhatRuntimeEnvironment } from "hardhat/types"
+import type { DeployFunction } from "hardhat-deploy/types"
+
+const func: DeployFunction = async function (hre: HardhatRuntimeEnvironment) {
+  const { ethers, getNamedAccounts, helpers, deployments } = hre
+  const { log } = deployments
+  const { deployer } = await getNamedAccounts()
+
+  // These are the fake random addresses for local development purposes only.
+  const fakeTokenBridge = "0x0af5DC16568EFF2d480a43A77E6C409e497FcFb9"
+  const fakeWormholeTBTC = "0xe1F0b28a3518cCeC430d0d86Ea1725e6256b0296"
+
+  const polygonTokenBridge = await deployments.getOrNull("PolygonTokenBridge")
+  const polygonWormholeTBTC = await deployments.getOrNull("PolygonWormholeTBTC")
+
+  const polygonTBTC = await deployments.get("PolygonTBTC")
+
+  let polygonTokenBridgeAddress = polygonTokenBridge?.address
+  if (!polygonTokenBridgeAddress && hre.network.name === "hardhat") {
+    polygonTokenBridgeAddress = fakeTokenBridge
+    log(`fake Polygon TokenBridge address ${polygonTokenBridgeAddress}`)
+  }
+
+  let polygonWormholeTBTCAddress = polygonWormholeTBTC?.address
+  if (!polygonWormholeTBTCAddress && hre.network.name === "hardhat") {
+    polygonWormholeTBTCAddress = fakeWormholeTBTC
+    log(`fake Polygon WormholeTBTC address ${polygonWormholeTBTCAddress}`)
+  }
+
+  const [, proxyDeployment] = await helpers.upgrades.deployProxy(
+    "PolygonWormholeGateway",
+    {
+      contractName:
+        "@keep-network/tbtc-v2/contracts/l2/L2WormholeGateway.sol:L2WormholeGateway",
+      initializerArgs: [
+        polygonTokenBridgeAddress,
+        polygonWormholeTBTCAddress,
+        polygonTBTC.address,
+      ],
+      factoryOpts: { signer: await ethers.getSigner(deployer) },
+      proxyOpts: {
+        kind: "transparent",
+      },
+    }
+  )
+
+  // TODO: Investigate the possibility of adding Tenderly verification for the
+  // sidechain and upgradable proxy.
+
+  // Contracts can be verified on Polygonscan in a similar way as we
+  // do it on L1 Etherscan
+  if (hre.network.tags.polygonscan) {
+    // Polygonscan might not include the recently added proxy transaction right
+    // after deployment. We need to wait some time so that transaction is
+    // visible on Polygonscan.
+    await new Promise((resolve) => setTimeout(resolve, 10000))
```

Comment:
Same question as for the token deployment. I think it makes more sense to have the verification in a separate step if the verification plugin can't handle the timing drift issue.

Reply:
Same as above. [accb791](https://github.com/keep-network/tbtc-v2/pull/588/commits/accb79171df01922f2a95c6f73e660f4966d56da)